modules/lib/src/main/resources/assets/js/app/browse/ContentBrowsePanel.ts

```diff
@@ -296,14 +297,51 @@ export class ContentBrowsePanel
         const path: string = this.getPathFromInlinePath(contentInlinePath);
 
         if (path) {
-            // possibly don't need this, but presumes expanding tree structure till the item is found
+            new ContentExistsByPathRequest(path).sendAndParse().then((exists: boolean) => {
+                const targetPath = ContentPath.create().fromString(path).build();
+                this.expandToListElementByPath(this.treeListBox, targetPath.getPathAtLevel(1), targetPath);
+            }).catch(DefaultErrorHandler.handle);
         }
     }
 
     private getPathFromInlinePath(contentPreviewPath: string): string {
         return UriHelper.getPathFromPortalInlineUri(contentPreviewPath, RenderingMode.INLINE);
     }
 
+    private expandToListElementByPath(list: ContentsTreeGridList, itemPath: ContentPath, targetPath: ContentPath): void {
+        let loadCount = 0;
+
+        const itemFinder = () => {
+            const listElement = list.getListElementByPath(itemPath)
+
+            if (listElement) {
+                list.unItemsAdded(itemFinder);
+
+                listElement.whenRendered(() => {
+                    listElement.getHTMLElement().scrollIntoView();
+
+                    if (itemPath.equals(targetPath)) {
+                        this.selectionWrapper.deselectAll();
+                        this.selectionWrapper.select(listElement.getItem());
+                    } else {
+                        listElement.expand(); // if was loaded but the children list collapsed
+                        this.expandToListElementByPath(listElement.getList(), targetPath.getPathAtLevel(itemPath.getLevel() + 1), targetPath);
```

Comment:
Same here, `targetPath.getPathAtLevel(itemPath.getLevel() + 1)` can be calculated once.

Reply:
same here, invoked just once